default.nix

```diff
@@ -1,8 +1,8 @@
-{ config, lib, pkgs, installShellFiles, ... }:
+{ buildGoModule, config, lib, pkgs, installShellFiles, version ? "latest" }:
```

Comment:
Quick question about this line.
There is a mechanism in PC which checks (on startup) if the user runs the latest version.
Basically, it compares the current version to the latest release in GitHub.
If PC will be compiled with the string "latest" instead of the real version, it will always show that there is a pending update.
Is there any way around this issue?

Reply:
Yup, great question.
So the point is you are now going to want to use the binary compiled by the `nixpkgs` repo from now on.

That will have the version available as defined in https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/misc/process-compose/default.nix

The package will show up here in a few days once the unstable channel "stabilizes":
https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=process-compose 

The only reason you would want to go through this way of building the software is to:
1. build a custom commit from source
2. use an unreleased version of the project (the above nixpkgs file is not recent enough)

In both cases you probably don't care about the newer versions, as this is an 'advanced' use-case.

I can see the purpose of the newer version check, but breaking it for 'custom' builds sounds acceptable to me.
Other software usually doesn't have this as prominent in the UI as `process-compose` has, I'm questioning the need for it as the software matures (`docker-compose` does not warn you about using an outdated version since years).

Reply:
I will start with the last part. The reason I added this feature is that I saw and liked a similar functionality in K9S. Also, it is very common (e.g. all the chromium based browsers, all the JetBrains IDEs, VsCode) to show this functionality in the UI.

If the issue is limited to custom builds, then I think that it is acceptable.
Is there a scenario in which I release a new version and someone who installed PC through nix will see this new version notification, but won't be able to update because there is a delay in updating the nix packages repository?
Will this repository be updated automatically or there is a manual process involved in bringing it to the main repo?

Reply:
> I will start with the last part. The reason I added this feature is that I saw and liked a similar functionality in K9S. Also, it is very common (e.g. all the chromium based browsers, all the JetBrains IDEs, VsCode) to show this functionality in the UI.

Sounds reasonable :tada: , I'm personally not using those programs daily, so I haven't been exposed to this pattern.

> Is there a scenario in which I release a new version and someone who installed PC through nix will see this new version notification, but won't be able to update because there is a delay in updating the nix packages repository?

Yes, this is quite common, as the updating frequency of the unstable channel is a few days usually.
This is not a unique problem to Nix though, image if Debian stable packaged PC and you'd do weekly releases.
Users would see constantly the out-dated warning in the TUI. 
VSCode/Chromium have self-updating capabilities to resolve these issues, although they beat the atomic upgrade features of package managers.
Tradeoffs, tradeoffs, tradeoffs..

> Will this repository be updated automatically or there is a manual process involved in bringing it to the main repo?

It's a manual effort. As the package maintainer in the `nixpkgs` repo, I'm responsible to keep updating the version/dependency hashes. This can be done by anyone though, taking k9s in nixpkgs as an example:
https://github.com/NixOS/nixpkgs/commits/master/pkgs/applications/networking/cluster/k9s/default.nix

Reply:
How about this:
Nix users won't care about this notification in any case because of 2 reasons:
1. There is nothing to do about it.
2. PC will be updated automatically when they run the system update command (I believe this one, right? :))

Github downloaders will still get this feature.

The way to distinguish between the 2 is to use different ldflags in the build process.
In my `Makefile` I will set the `version` variable and in `default.nix` I will set the `nixVersion` variable. The `version` will remain `undefined`, which will let me know not to pull the latest release from GitHub.

What do you think?

Reply:
I like the idea!

I would just replace `Nix` with `package manager` users in general in your comment.

> In my Makefile I will set the version variable and in default.nix I will set the nixVersion variable. The version will remain undefined, which will let me know not to pull the latest release from GitHub.

No need for setting `nixVersion` specific ldflag. Instead let's add an auto-update check specific boolean flag.
This way people who are source-builders and don't want network requests on TUI startup to happen can opt-out of it.
I'll modify the packaging source code in nixpkgs to set this to false (and also in this PR).

Does that sound okay?

Reply:
Perfect!